examples/basic_qa_pipeline.py

```diff
@@ -0,0 +1,76 @@
+import logging
+from pathlib import Path
+
+logging.basicConfig(format="%(levelname)s - %(name)s -  %(message)s", level=logging.WARNING)
+logging.getLogger("haystack").setLevel(logging.INFO)
+
+from haystack.document_stores import ElasticsearchDocumentStore
+from haystack.utils import clean_wiki_text, convert_files_to_docs, fetch_archive_from_http, print_answers, launch_es
+from haystack.nodes import FARMReader, BM25Retriever
+from haystack.nodes.file_classifier import FileTypeClassifier
+from haystack.nodes.preprocessor import PreProcessor
+from haystack.nodes.file_converter import TextConverter
+from haystack.pipelines import Pipeline
+from pprint import pprint
+
+
+def basic_qa_pipeline():
+    # launch and create DocumentStore
+    launch_es()
+
+    # Initialize a DocumentStore
+    document_store = ElasticsearchDocumentStore(host="localhost", username="", password="", index="document")
+
+    # fetch, pre-process and write documents
+    doc_dir = "data/basic_qa_pipeline"
+    s3_url = "https://s3.eu-central-1.amazonaws.com/deepset.ai-farm-qa/datasets/documents/wiki_gameofthrones_txt1.zip"
+    fetch_archive_from_http(url=s3_url, output_dir=doc_dir)
+
+    file_paths = [p for p in Path(doc_dir).glob("**/*")]
+    files_metadata = [{"name": path.name} for path in file_paths]
+
+    # Indexing Pipeline
+    indexing_pipeline = Pipeline()
+
+    # Makes sure the file is a TXT file (FileTypeClassifier node)
+    classifier = FileTypeClassifier()
+    indexing_pipeline.add_node(classifier, name="Classifier", inputs=["File"])
+
+    # Converts a file into text and performs basic cleaning (TextConverter node)
+    text_converter = TextConverter(remove_numeric_tables=True)
+    indexing_pipeline.add_node(text_converter, name="Text_converter", inputs=["Classifier.output_1"])
+
+    # - Pre-processes the text by performing splits and adding metadata to the text (Preprocessor node)
+    preprocessor = PreProcessor(
+        clean_whitespace=True,
+        clean_empty_lines=True,
+        split_length=100,
+        split_overlap=50,
+        split_respect_sentence_boundary=True,
+    )
+    indexing_pipeline.add_node(preprocessor, name="Preprocessor", inputs=["Text_converter"])
+
+    # - Writes the resulting documents into the document store
+    indexing_pipeline.add_node(document_store, name="Document_Store", inputs=["Preprocessor"])
+
+    # Then we run it with the documents and their metadata as input
+    indexing_pipeline.run(file_paths=file_paths, meta=files_metadata)
+
+    # Initialize Retriever & Reader
+    retriever = BM25Retriever(document_store=document_store)
```

Comment:
Not a blocker, but shall we use a dense retriever in here? Just because I often need to go find our default sentence-transformers HF name, having it here would be handy :grin: 

```python
retriever = EmbeddingRetriever(
    document_store=document_store,
    embedding_model="sentence-transformers/multi-qa-mpnet-base-dot-v1",
    model_format="sentence_transformers",
)
```

Reply:
I suppose it's easier to switch the code from EmbeddingRetriever -> BM25 rather than it being the other way?

Reply:
Ok, I've made this change but I have not pushed it because now the script takes ages to run. I will leave you to make the final call. Add or keep as BM25?

Reply:
> I suppose it's easier to switch the code from EmbeddingRetriever -> BM25 rather than it being the other way?

Exactly, that's the main reason.

> Ok, I've made this change but I have not pushed it because now the script takes ages to run. I will leave you to make the final call. Add or keep as BM25?

Maybe we can use a smaller, toy dataset then? This is just an example script, no need to index plenty of files.

Reply:
I was thinking about something drastically smaller, like one single page from Wikipedia. After all these scripts are just for the code, we don't really care about the dataset right?

When we had the CI workflow in OSS we used to run them on a very small subset of the GoT dataset. If I recall correctly we were taking the first 10 files or something like that. My guess is that we can go even smaller with no issues.